hive-metastore/src/main/java/org/apache/iceberg/hive/HiveSchemaUtil.java

```diff
@@ -0,0 +1,183 @@
+    fieldSchemas.forEach(fieldSchema -> builder.identity(fieldSchema.getName()));
+    return builder.build();
+  }
+
+  /**
+   * Converts the Hive list of column names and column types to an Iceberg schema. If some of the types are not
+   * convertible then exception is thrown.
+   * @param names The list of the Hive column names
+   * @param types The list of the Hive column types
+   * @param comments The list of the Hive column comments
+   * @return The Iceberg schema
+   */
+  public static Schema convert(List<String> names, List<TypeInfo> types, List<String> comments) {
+    return HiveSchemaConverter.convert(names, types, comments, false);
+  }
+
+  /**
+   * Converts the Hive list of column names and column types to an Iceberg schema.
+   * @param names The list of the Hive column names
+   * @param types The list of the Hive column types
+   * @param comments The list of the Hive column comments, can be null
+   * @param autoConvert If <code>true</code> then TINYINT and SMALLINT is converted to INTEGER and VARCHAR and CHAR is
+   *                    converted to STRING. Otherwise if these types are used in the Hive schema then exception is
+   *                    thrown.
+   * @return The Iceberg schema
+   */
+  public static Schema convert(List<String> names, List<TypeInfo> types, List<String> comments, boolean autoConvert) {
+    return HiveSchemaConverter.convert(names, types, comments, autoConvert);
+  }
+
+  /**
+   * Converts an Iceberg type to a Hive TypeInfo object.
+   * @param type The Iceberg type
+   * @return The Hive type
+   */
+  public static TypeInfo convert(Type type) {
+    return TypeInfoUtils.getTypeInfoFromTypeString(convertToTypeString(type));
+  }
+
+  /**
+   * Converts a Hive typeInfo object to an Iceberg type.
+   * @param typeInfo The Hive type
+   * @return The Iceberg type
+   */
+  public static Type convert(TypeInfo typeInfo) {
+    return HiveSchemaConverter.convert(typeInfo, false);
+  }
+
+  private static String convertToTypeString(Type type) {
+    switch (type.typeId()) {
+      case BOOLEAN:
+        return "boolean";
+      case INTEGER:
+        return "int";
+      case LONG:
+        return "bigint";
+      case FLOAT:
+        return "float";
+      case DOUBLE:
+        return "double";
+      case DATE:
+        return "date";
+      case TIME:
+      case STRING:
+      case UUID:
+        return "string";
+      case TIMESTAMP:
+        Types.TimestampType timestampType = (Types.TimestampType) type;
+        if (MetastoreUtil.hive3PresentOnClasspath() && timestampType.shouldAdjustToUTC()) {
+          return "timestamp with local time zone";
+        }
+        return "timestamp";
+      case FIXED:
+      case BINARY:
+        return "binary";
+      case DECIMAL:
+        final Types.DecimalType decimalType = (Types.DecimalType) type;
+        return String.format("decimal(%s,%s)", decimalType.precision(), decimalType.scale());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;precision&#39; expression, &#39;scale&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 166. Consider formatting this data with &#39;%d&#39; instead.